roles/postgres-cluster-pgbouncer-client-setup/tasks/main.yml

```diff
@@ -1,28 +1,51 @@
 ---
+- name: pull out scram-sha-256 hashes for building out userlist.txt
+  ansible.builtin.shell:
+    psql -p "{{ postgresql_cluster_port }}" -t -X -A -c "select rolpassword from pg_authid where rolname = '{{ item.name }}'"
+  become_user: "{{ postgresql_cluster_user }}"
+  loop: "{{ postgresql_cluster_users | default([]) | flatten(levels=1) }}"
+  when: not (postgresql_cluster_is_monitor | default('False') | bool) and (inventory_hostname == play_hosts[1])
```

Comment:
is play_hosts[1] the first host or the second?

Reply:
This runs on af-database-01 (af-monitor-01 is play_hosts[0]). This has always been the active node on deployment for me. Both nodes should be read-only anyway (but the monitor will not work for sure). The goal is to only run on one database node as it was easier for me to implement this solution than it was to try and deal with pulling the results and pushing the template to each machine individually using the previous method that was implemented when it was just md5 which was easily supported in Ansible.

Reply:
Okay, I understand that, but I'd prefer for this to not be decided by an index in play_hosts. Instead we should get the first non manager node instead by filtering the list of hosts in the inventory by postgresql_cluster_is_monitor and then using the first element in that list.

Reply:
Also, play_hosts seems to be deprecated:

https://docs.ansible.com/ansible/latest/reference_appendices/special_variables.html#term-play_hosts